dd-sdk-android-core/src/main/kotlin/com/datadog/android/core/internal/receiver/ThreadSafeReceiver.kt

```diff
@@ -18,13 +18,15 @@ internal abstract class ThreadSafeReceiver : BroadcastReceiver() {
 
     val isRegistered = AtomicBoolean(false)
 
-    @SuppressLint("UnspecifiedRegisterReceiverFlag")
+    @SuppressLint("WrongConstant", "UnspecifiedRegisterReceiverFlag")
     fun registerReceiver(
         context: Context,
         filter: IntentFilter
     ): Intent? {
         val intent = if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU) {
             context.registerReceiver(this, filter, Context.RECEIVER_NOT_EXPORTED)
+        } else if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
```

Comment:
Normally we should use ContextCompat here:
https://developer.android.com/reference/androidx/core/content/ContextCompat#RECEIVER_NOT_EXPORTED() 
but not sure if we should bring the ContextCompat dependency only for this reason. The drawback on this approach would be in case that Flag value will be changed in the future (quite unlikely) . I am not sure we should use ContextCompat  only for this reason. What do you think ?